packages/validation/src/validateField/conditions/testConditionalRestriction.ts

```diff
@@ -0,0 +1,130 @@
+ */
+
+import {
+	ARRAY_TEST_CASE_DEFAULT,
+	ArrayTestCase,
+	type ArrayDataValue,
+	type ConditionalRestrictionTest,
+	type DataRecord,
+	type DataRecordValue,
+	type RestrictionCondition,
+	type SingleDataValue,
+} from '@overture-stack/lectern-dictionary';
+import { resultForArrayTestCase } from '../../utils/resultForArrayTestCase';
+import { testMatchCount } from './testMatchCount';
+import { testMatchCodeList } from './testMatchCodeList';
+import { testMatchExists } from './testMatchExists';
+import { testMatchRange } from './testMatchRange';
+import { testMatchRegex } from './testMatchRegex';
+import { testMatchValue } from './testMatchValue';
+
+/**
+ * Test values extracted from other fields vs a match test. This function should be passed the
+ *
+ * @param values
+ * @param rule
+ * @param matchTest
+ * @param arrayCase  Intentionally using `| undefined` vs `/?: arrayCase` to ensure that this argument is provided when this function is called
+ * @returns
+ */
+const fieldsPassMatchTest = (
+	values: DataRecordValue[],
+	matchTest: (value: DataRecordValue) => boolean,
+	arrayCase: ArrayTestCase | undefined,
+): boolean => {
+	const fieldTestResults = values.map((value) => matchTest(value));
+	return resultForArrayTestCase(fieldTestResults, arrayCase || ARRAY_TEST_CASE_DEFAULT);
+};
+
+const testConditionForSingularValue = (
+	condition: RestrictionCondition,
+	_value: SingleDataValue,
+	fieldValues: DataRecordValue[],
+): boolean => {
+	const matchCodeList = condition.match.codeList;
+	if (matchCodeList !== undefined) {
+		if (!fieldsPassMatchTest(fieldValues, (value) => testMatchCodeList(matchCodeList, value), condition.case)) {
+			return false;
+		}
+	}
+	const matchCount = condition.match.count;
+	if (matchCount !== undefined) {
+		if (!fieldsPassMatchTest(fieldValues, (value) => testMatchCount(matchCount, value), condition.case)) {
+			return false;
+		}
+	}
+
+	const matchExists = condition.match.exists;
+	if (matchExists !== undefined) {
+		if (!fieldsPassMatchTest(fieldValues, (value) => testMatchExists(matchExists, value), condition.case)) {
+			return false;
+		}
+	}
+
+	const matchRange = condition.match.range;
+	if (matchRange !== undefined) {
+		if (!fieldsPassMatchTest(fieldValues, (value) => testMatchRange(matchRange, value), condition.case)) {
+			return false;
+		}
+	}
+
+	const matchRegex = condition.match.regex;
+	if (matchRegex !== undefined) {
+		if (!fieldsPassMatchTest(fieldValues, (value) => testMatchRegex(matchRegex, value), condition.case)) {
+			return false;
+		}
+	}
+
+	const matchValue = condition.match.value;
+	if (matchValue !== undefined) {
+		if (!fieldsPassMatchTest(fieldValues, (value) => testMatchValue(matchValue, value), condition.case)) {
+			return false;
+		}
+	}
+	return true;
+};
+
+const testConditionForArray = (
+	_condition: RestrictionCondition,
+	_value: ArrayDataValue,
+	_fieldValues: DataRecordValue[],
+): boolean => {
+	throw new Error('Unimplemented.');
```

Comment:
what would be the case when this function returns on error? when raw data comes as an array format `['apple', 'strawberry']` or when is comma separated `apple,strawberry`?

Reply:
The original values from a tsv would all be strings like `"apple,strawberry`". There are two steps to process the data, `parse` and `validate`. During parsing the value would be converted to `['apple', 'strawberry']`.